src/Microsoft.AspNet.Mvc.Core/ModelBinding/GenericModelBinder.cs

```diff
@@ -13,12 +13,21 @@ public class GenericModelBinder : IModelBinder
     {
         public async Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
         {
-            var binderType = ResolveBinderType(bindingContext.ModelType);
+            var binderType = ResolveBinderType(bindingContext);
             if (binderType != null)
             {
                 var binder = (IModelBinder)Activator.CreateInstance(binderType);
-                var result = await binder.BindModelAsync(bindingContext);
 
+                var collectionBinder = binder as ICollectionModelBinder;
+                if (collectionBinder != null &&
+                    bindingContext.Model == null &&
+                    collectionBinder.CreateEmptyCollection(bindingContext.ModelType) == null)
```

Comment:
Rather than this, it would be nice to add a `CanBindType` or similar to `ICollectionModelBinder`.

This also seems to ignore cases where we can safely update a type, but can't construct it.